### PR TITLE
Upgrade to Phoenix 1.4

### LINKED ISSUE #18: Upgrade to Phoenix 1.4
Looks like it could be relatively easy.

- Jason is default json parser (might be good to swap poison) (a couple config/endpoint changes).
- Some modifications required for ecto_sql
- Seemed straight-forward w your tests already in place.

### PR DESCRIPTION
**Pull request by JohnKacz**
Should close #18 

Disclaimer: I'm not exactly sure what I'm doing. I just followed the upgrade guide and then hunted down some failing tests.

### CODE REVIEW

File: example/assets/yarn.lock [hunk 1]
```diff
@@ -1,2761 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (jeremyjh) [inline example/assets/yarn.lock]: Any particular reason to switch from yarn to npm lock files in the example?

File: mix.exs [hunk 2]
```diff
@@ -29,7 +29,8 @@ defmodule PhoenixDatatables.Mixfile do
   # Type `mix help deps` for examples and options.
   defp deps do
     [
-      {:ecto, "~> 2.1"},
+      # {:ecto, "~> 2.1"},
```
Comment (jeremyjh) [inline mix.exs]: We should just remove it rather than comment it.

File: example/assets/yarn.lock (hunk 1 above)
Comment (JohnKacz) [reply example/assets/yarn.lock]: Nope. I assumed it was connected to the old brunch workflow and removed it along with `example/assets/brunch-config.js`

Comment (jeremyjh) [reply example/assets/yarn.lock]: Its not related to brunch. Does mix phx.new create an npm lockfile though? I guess we can switch to it if it does.

Comment (JohnKacz) [reply example/assets/yarn.lock]: yes `mix phx.new myapp` makes a file `myapp/assets/package-lock.json`

**Comment by JohnKacz** [comment]
Oh probably also need to update the README as it mentions `Poison`

**Comment by jeremyjh** [comment]
> Oh probably also need to update the README as it mentions `Poison` and brunch files

Yeah I'd just use the README it would generate with 1.4's `mix phx.new` in the example.

**Comment by JohnKacz** [comment]
> Yeah I'd just use the README it would generate with 1.4's `mix phx.new` in the example.

Ah ok I replaced that for the example README, but I was referring to the project's README. Specifically in the [View](https://github.com/smartmetals/phoenix_datatables/blob/master/README.md#view) section.

And looking a little further down the instructions for Client reference `brunch-config.js` which should be changed.

All that led me to realize that `webpack.config.js` is currently not setup correctly. It's just the generated default file, no datatables configuration. So the Item's index page doesn't load the table.

I'm still learning webpack so if you have advice I'd be happy to take it. Otherwise I'll keep looking into how to set that up properly and updating the project README.

File: example/lib/phoenix_datatables_example_web/templates/item/index.html.eex [hunk 3]
```diff
@@ -10,8 +10,9 @@
       <th>Price</th>
       <th>Ui</th>
       <th>Aac</th>
+      <th></th>
```
Comment (JohnKacz) [inline example/lib/phoenix_datatables_example_web/templates/item/index.html.eex]: This should not have been added.

**Comment by JohnKacz** [comment]
> All that led me to realize that `webpack.config.js` is currently not setup correctly. It's just the generated default file, no datatables configuration. So the Item's index page doesn't load the table.
> 
> I'm still learning webpack so if you have advice I'd be happy to take it. Otherwise I'll keep looking into how to set that up properly and updating the project README.

In case it helps the error I'm getting from calling `dt();` [here](https://github.com/smartmetals/phoenix_datatables/blob/master/example/assets/js/table.js#L6) is:

````
jQuery.Deferred exception: this is undefined DataTable@webpack:///./node_modules/datatables.net/js/jquery.dataTables.js?:112:3
__webpack_exports__.default/<@webpack:///./js/table.js?:10:5
mightThrow@webpack:///./node_modules/jquery/dist/jquery.js?:3534:21
resolve/</process<@webpack:///./node_modules/jquery/dist/jquery.js?:3602:12
 undefined
`````

**Comment by jeremyjh** [comment]
I think we no longer need to call `dt();` in `tables.js` if we're using webpack. I didn't try it, but I noticed in the examples from this thread people were not doing it.  https://datatables.net/forums/discussion/32542/datatables-and-webpack

**Comment by JohnKacz** [comment]
> I think we no longer need to call `dt();` in `tables.js` if we're using webpack.

That worked! I updated the example project's README to the new generated default (some links were updated) and just removed the sections referring to brunch.

I think that does it. What do you think?

**Comment by jeremyjh** [comment]
Hey, thanks for plugging away at this. I agree it looks like we are really close. One thing I think is missing are the Datatables images. I don't see the little sort arrows above the columns. These images are copied with a brunch plugin 'copycat', per `brunch-config.js`.

```javascript
    copycat: {
      'images': [ 'node_modules/datatables.net-dt/images' ],
    },

```

I Googled a bit and there is an equivalent webpack module in contrib: https://github.com/webpack-contrib/copy-webpack-plugin .

Webpack also supports inclusion of static files via `require` but to me that would make the most sense if the datatables NPM package were requiring both CSS and images. Since it doesn't, I think the plugin is a good workaround.

**Comment by jeremyjh** [comment]
I think we need to also include the Datatables CSS. I'm not sure the best way to do that, in some of the examples they do use require for that.

**Comment by JohnKacz** [comment]
I'm unsure of what to do here. 

I tried adding `{ from: 'node_modules/datatables.net-dt/images', to: 'static/images' },` to the CopyWebpackPlugin, but I don't know if that is working because the CSS isn't loaded. 

I tried a few things to load/bundle the the `jquery.datatables.css` CSS but I clearly don't know what I'm doing 😜

**Comment by jeremyjh** [comment]
Hey I managed to get this working. We actually don't need to use the CopyWebpackPlugin for this. Instead we'll import the CSS in app.js and url-loader will embed the images in a data-uri. 

```
npm install url-loader --save-dev
```


```javascript
//app.js
import _ from "datatables.net-dt/css/jquery.datatables.css"
```

```javascript
//webpack.config.js
module: {
  rules: [
{ test: /\.png$/, loader: "url-loader?mimetype=image/png" }
```

**Comment by JohnKacz** [comment]
Awesome. This looks to be working for me as well. Thanks for the help.

**Comment by jeremyjh** [comment]
Thanks! Sorry it took so long to get back this. I pushed the udpated library to Hex as 0.3.0.

**Pull request merged**